bitcoin/src/crypto/sighash.rs

```diff
@@ -907,18 +907,18 @@ impl<R: Borrow<Transaction>> SighashCache<R> {
     ///
     /// This function can't handle the SIGHASH_SINGLE bug internally, so it returns [`EncodeSigningDataResult`]
     /// that must be handled by the caller (see [`EncodeSigningDataResult::is_sighash_single_bug`]).
-    pub fn legacy_encode_signing_data_to<W: Write + ?Sized, U: Into<u32>>(
+    pub fn legacy_encode_signing_data_to<W: Write + ?Sized>(
         &self,
         writer: &mut W,
         input_index: usize,
         script_pubkey: &Script,
-        sighash_type: U,
+        sighash_type: EcdsaSighashType,
```

Comment:
IIRC this was needed to dump shitcoins which use non-standard type. While we don't support shitcoins, maybe allowing to dump them wouldn't be too bad? Though the function should just accept `u32` and it should have some scary name.

Reply:
It's probably not the end of the world to require people use an old version of rust-bitcoin or to copy/paste this method in order to write shitcoin-dumping code.

This was really only useful in the immediate aftermath of the bcash fork (and then again, at like 1/10th the yield), after BSV. It would surprise me if this happened again in the same way and it's probably fine to no longer support it in the API.

Reply:
Oh that's great news! I really didn't like it even though it made sense. ACK the change then.

Reply:
Also if we ever need to do anything like that again I propose to just copy paste the code into a separate crate and tell people to use it instead.